addons/pos_self_order/controllers/controllers.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+import werkzeug
+from odoo import http
+from odoo.http import request
+from typing import List, Dict, Optional
+from odoo.addons.http_routing.models.ir_http import unslug
+
+
+class PosSelfOrder(http.Controller):
+    """
+    This is the controller for the POS Self Order App
+    There is one main route that the clientn will use to access the POS Self Order App: /menu
+    This route will render the LANDING PAGE of the POS Self Order App
+    And it will pass some generic variabiles to the template: pos_id, table_id, company_name, currency...
+    After that the client will be able to navigate to the /products route w/o aditional requests
+    to the server, using client side routing.
+    """
+
+    @http.route("/menu", auth="public", website=True)
+    def pos_self_order_redirect(self):
+        pos_id = find_any_pos_with_qr_code_menu_sudo().id
+        return request.redirect(f"/menu/{pos_id}")
+
+    @http.route(
+        [
+            "/menu/<string:pos_name>",
+            "/menu/<string:pos_name>/products",
+            "/menu/<string:pos_name>/products/<int:product_id>",
+        ],
+        auth="public",
+        website=True,
+    )
+    def pos_self_order_start(self, pos_name, product_id: int = None, auth: str = None):
+        """
+        The user gets this route from the QR code that they scan at the table
+        This START ROUTE will render the POS Self Order App
+        And it will pass some generic variabiles to the template: pos_id, table_id, company_name, currency,
+        We get some details about this POS from the model "pos.config"
+        """
+        _, pos_id = unslug(pos_name)
```

Comment:
what do you think about this:
```
        """
        The user gets this route from the QR code that they scan at the table
        This START ROUTE will render the POS Self Order App
        And it will pass the needed data to the template: the product list, pos_config_id, company_name, currency, etc.
        :param pos_name: the name of the pos config: can be the id or the slugified name of the pos config
        :type pos_name: str
        :return: the rendered template
        """
```